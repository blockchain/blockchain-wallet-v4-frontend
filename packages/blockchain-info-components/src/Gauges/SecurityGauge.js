import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Color } from '../Colors/index.ts'
import { Icon } from '../Icons'

const Empty = styled.div`
  height: 13px;
  width: 13px;
  border: 1px solid red;
  border-radius: 50%;
  margin-right: 5px;
`
const Progress = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 60px;
  padding-left: 5px;
`
const SecurityIcon = styled(Icon)`
  border-radius: 50%;
  background: ${Color('success')};
  height: 16px;
  width: 15px;
  color: white;
  cursor: inherit;
  margin-right: 5px;
`

const SecurityGauge = props => {
  const { score } = props
  const hide = score === 3

  if (hide) return null
  return (
    <Progress>
      {score >= 1 ? (
        <SecurityIcon name='checkmark-in-circle' size='16px' />
      ) : (
        <Empty />
      )}
      {score >= 2 ? (
        <SecurityIcon name='checkmark-in-circle' size='16px' />
      ) : (
        <Empty />
      )}
      {score === 3 ? (
        <SecurityIcon name='checkmark-in-circle' size='16px' />
      ) : (
        <Empty />
      )}
    </Progress>
  )
}

SecurityGauge.propTypes = {
  score: PropTypes.oneOf([0, 1, 2, 3]).isRequired
}

SecurityGauge.defaultProps = {
  score: 0
}

export default SecurityGauge
