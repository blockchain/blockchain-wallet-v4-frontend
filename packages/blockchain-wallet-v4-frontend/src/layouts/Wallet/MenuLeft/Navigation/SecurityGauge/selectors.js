import { selectors } from 'data'
import { sequence, reduce, add, prop } from 'ramda'
import { Remote } from 'blockchain-wallet-v4/src'

export const getData = (state) => {
  const settings = [
    selectors.core.settings.getEmailVerified(state),
    selectors.core.wallet.isMnemonicVerified(state) ? Remote.of(1) : Remote.of(0),
    prop('data', selectors.core.settings.getAuthType(state)) > 0 ? Remote.of(1) : Remote.of(0)
  ]
  return sequence(Remote.of, settings).map(reduce(add, 0))
}
