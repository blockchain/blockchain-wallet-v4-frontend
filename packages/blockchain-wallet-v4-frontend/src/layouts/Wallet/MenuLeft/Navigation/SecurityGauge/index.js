import React from 'react'
import ui from 'redux-ui'
import { compose, bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { actions } from 'data'
import { getData } from './selectors'
import { SecurityGauge } from 'blockchain-info-components'

class SecurityGaugeContainer extends React.PureComponent {
  render () {
    const { data } = this.props

    return data.cata({
      Success: value => <SecurityGauge score={value} />,
      Failure: message => <div />,
      Loading: () => <div />,
      NotAsked: () => <div />
    })
  }
}

const mapStateToProps = state => ({
  data: getData(state)
})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions.core.settings, dispatch)
})

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  ui({ key: 'MenuLeft', persist: true, state: { settingsToggled: false } })
)

export default enhance(SecurityGaugeContainer)
