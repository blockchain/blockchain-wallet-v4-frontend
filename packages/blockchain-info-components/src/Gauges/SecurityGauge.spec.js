import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SecurityGauge from './SecurityGauge'

describe('SecurityGauge component', () => {
  it('default renders correctly', () => {
    const component = shallow(<SecurityGauge score={1} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
