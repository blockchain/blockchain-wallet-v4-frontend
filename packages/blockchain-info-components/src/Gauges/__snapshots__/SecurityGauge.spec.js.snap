// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityGauge component default renders correctly 1`] = `
<SecurityGauge__Progress>
  <SecurityGauge__SecurityIcon
    name="checkmark-in-circle"
    size="16px"
  />
  <SecurityGauge__Empty />
  <SecurityGauge__Empty />
</SecurityGauge__Progress>
`;
